src/UI/state/upload/uploadSlice.tsx

```diff
@@ -1,20 +1,68 @@
 import { createSlice } from '@reduxjs/toolkit';
 import { getTemplateList } from './actions/downloadTemplate';
 
+export type ErrorRow = {
+  row: number;
+  data: {
+    key: string;
+    errorType: string;
+    expectedType?: string;
+    receivedType?: string;
+  }[];
+};
+
 export interface UploadState {
   modelFile: File | null;
   dataFile: File | null;
   loading: boolean;
+  validationErrors: ErrorRow[];
   templateList: string[];
 }
 
 export const initialState: () => UploadState = () => ({
   modelFile: null,
   dataFile: null,
   loading: false,
+  validationErrors: [],
   templateList: [],
 });
 
+export function unpackOverlays(map_layers: any) {
```

Comment:
What's this doing here?

Reply:
Oh I know what I've done. I used it for reference and have forgotten to remove it